tests/test_models.py

```diff
@@ -1,341 +1,122 @@
-@pytest.mark.parametrize(
-    'which_file, skip_fits_update, expected_exp_type',
-    [
-        ('just_fits', None,  'FGS_DARK'),
-        ('just_fits', False, 'FGS_DARK'),
-        ('just_fits', True,  'FGS_DARK'),
-        ('model',     None,  'FGS_DARK'),
-        ('model',     False, 'FGS_DARK'),
-        ('model',     True,  'NRC_IMAGE')
-    ]
-)
-# @pytest.mark.parametrize(
-#     'open_func',
-#     [DataModel, datamodels.open]
-# )
-@pytest.mark.parametrize(
-    'use_env',
-    [False, True]
-)
-def test_skip_fits_update(jail_environ,
-                          use_env,
-                          make_models,
-                          open_func,
-                          which_file,
-                          skip_fits_update,
-                          expected_exp_type):
-    """Test skip_fits_update setting"""
-    # Setup the FITS file, modifying a header value
-    path = make_models[which_file]
-    hduls = fits.open(path)
-    hduls[0].header['exp_type'] = 'FGS_DARK'
-
-    # Decide how to skip. If using the environmental,
-    # set that and pass None to the open function.
-    try:
-        del os.environ['SKIP_FITS_UPDATE']
-    except KeyError:
-        # No need to worry, environmental doesn't exist anyways
-        pass
-    if use_env:
-        if skip_fits_update is not None:
-            os.environ['SKIP_FITS_UPDATE'] = str(skip_fits_update)
-            skip_fits_update = None
-
-    model = open_func(hduls, skip_fits_update=skip_fits_update)
-    assert model.meta.exposure.type == expected_exp_type
+    assert model.meta.telescope == "crystal ball"
 
 
-@pytest.mark.skip("requires ImageModel")
 def test_set_shape():
-    with ImageModel((50, 50)) as dm:
+    with BasicModel((50, 50)) as dm:
         assert dm.shape == (50, 50)
 
         with pytest.raises(AttributeError):
             dm.shape = (42, 23)
 
 
-@pytest.mark.skip("requires ImageModel")
 def test_broadcast():
-    with ImageModel((50, 50)) as dm:
+    with BasicModel((50, 50)) as dm:
         data = np.empty((50,))
         dm.dq = data
 
 
-@pytest.mark.skip("requires ImageModel")
 def test_broadcast2():
-    with ImageModel() as dm:
+    with BasicModel() as dm:
         data = np.empty((52, 50))
         dm.data = data
 
         dq = np.empty((50,))
         dm.dq = dq
 
 
-@pytest.mark.skip("requires jwst model implementations")
-def test_from_hdulist():
-    from astropy.io import fits
-    warnings.simplefilter("ignore")
-    with fits.open(FITS_FILE, memmap=False) as hdulist:
-        with datamodels.open(hdulist) as dm:
-            dm.data
-        assert not hdulist.fileinfo(0)['file'].closed
-
-
 def delete_array():
-    with datamodels.open() as dm:
-        del dm.data
-
-
-@pytest.mark.skip("base model no longer includes attribute meta.subarray")
-def test_subarray():
-    with DataModel(FITS_FILE) as dm:
-        dm.meta.subarray.xstart
-
+    with BasicModel() as m:
+        del m.data
```

Comment:
What does this do?  It's not a test and doesn't seem to be called by anything.

Reply:
I imagine at some point in history there was an exception when you deleted an ndarray field?

Reply:
Peharps.  It can be removed now.